pallets/pool-system/src/tranches.rs

```diff
@@ -1051,41 +1174,41 @@ where
 	pub fn non_residual_tranches(
 		&self,
 	) -> Option<&[EpochExecutionTranche<Balance, BalanceRatio, Weight, TrancheCurrency>]> {
-		if let Some((_head, tail)) = self.tranches.deref().as_slice().split_first() {
-			Some(tail)
-		} else {
-			None
-		}
+		self.tranches
+			.deref()
+			.as_slice()
+			.split_first()
+			.map(|(_head, tail)| tail)
+			.filter(|tail| !tail.len().is_zero())
```

Comment:
We are a little inconsistent here, if I am not mistaken. 
For the `struct Tranches` we have a better check than `split_first`. Maybe adapt the `EpochExecutionTranches` to behave the same.

Reply:
Ah you are right. 

Mmh. Want to get rid of the epoch tranches anyways. But let's keep it for now.

Reply:
Looking forward to getting rid of them! The code duplication strikes me.